File: frontend/src/administrator/bankAccount/service/BankAccountService.js
import http from '@/config/axios';

export default class BankAccountService {
    constructor() {
        this.endpoint = '/api/v1/bank-accounts';
    }

    findById(bankAccountId) {
        return http.get(`${this.endpoint}/${bankAccountId}`).then((response) => response);
    }

    findByFilter(bankAccountFilter) {
        return http.get(`${this.endpoint}/filter${bankAccountFilter.toString()}`).then((response) => response);
    }

    save(bankAccount) {
        if (bankAccount.id === null) {
            return http.post(`${this.endpoint}`, bankAccount).then((response) => response);
        }
        return this.#update(bankAccount);
    }

    #update(bankAccount) {
        return http.put(`${this.endpoint}`, bankAccount).then((response) => response);
    }
}

File: backend/bank-account/api/src/main/java/br/com/systec/fintrack/bankaccount/service/BankAccountService.java
package br.com.systec.fintrack.bankaccount.service;

import br.com.systec.fintrack.bankaccount.filter.BankAccountFilterVO;
import br.com.systec.fintrack.bankaccount.model.BankAccount;
import br.com.systec.fintrack.commons.exception.BaseException;
import br.com.systec.fintrack.commons.model.TransactionType;
import org.springframework.data.domain.Page;

public interface BankAccountService {

    BankAccount save(BankAccount bankAccount) throws BaseException;

    BankAccount update(BankAccount bankAccount) throws BaseException;

    BankAccount findById(Long id) throws BaseException;

    Page<BankAccount> findByFilter(BankAccountFilterVO filterVO) throws BaseException;

    Double findCurrentAccountBalance() throws BaseException;

    void updateBankAccountBalance(Double amount, Long bankAccountId, TransactionType transactionType) throws BaseException;
}
